static/app/utils/number/ladder.tsx

```diff
@@ -0,0 +1,63 @@
+import orderBy from 'lodash/orderBy';
+
+type Rung<T> = [threshold: number, value: T];
+
+/**
+ * Class that represents a value ladder. Given a ladder of increasing thresholds,
+ * the ladder can match an incoming value against the known intervals.
+ *
+ * @example
+ * ```javascript
+ * // Selecting a quota based on team size
+ * const ladder = new Ladder([
+ *   [0, 10],
+ *   [10, 100_000],
+ *   [30, 1_000_000],
```

Comment:
My interval tree suggestion is wrong, I have just realized that I have read this as intervals and not [range, value], sorry about that

Reply:
Does it make sense to change the API here to have a named property? When using [number, number], it isnt clear which one is the threshold and which one the rung value

Reply:
Yes, done!